Grammerly example (DO NOT MERGE) (#15130)
newrelic/docs-website: src/content/docs/new-relic-solutions/new-relic-one/ui-data/service-maps/uninstrumented-entities-service-map.mdx

---
import ServiceMapUninstrumentedEntity from 'images/mapsSolutions_screenshot-crop_uninstrumented-entity.webp'

New Relic can detect gaps on your instrumentation by analyzing your telemetry.These gaps can be databases, apps or basically any other type of entity. For example, one application could be storing information in a non-instrumented Amazon RDS database.

When that happens, you'll see in the form of an "uninstrumented" entity the created gaps. You can check it in the [service map](/docs/new-relic-solutions/new-relic-one/ui-data/service-maps/introduction-service-maps). You can also see the relationship between this uninstrumented entity and the instrumented one, which in this example is CALLS.

<Callout variant="tip">
  Note that uninstrumented and instrumented entities have different appearances. Uninstrumented entities use a dotted border while instrumented entities use a solid one.
</Callout>

> To instrument an uninstrumented entity, you need to click on it, choose `instrument`, and follow the guided tutorial on how to perform the instrumentations which depends on the type of entity. After you finish this process, the service map display the entities with a solid border.

Author: ```suggestion
To instrument an uninstrumented entity, you need to click on it, choose `instrument`, and follow the guided tutorial on how to perform the instrumentation which depends on the type of entity. After you finish this process, the service map displays the entities with a solid border.
```

> <figcaption>
> An example of an automap having the instrument option. 
> </figcaption>
> 
> Make sure that the entities are being monitored with an agent. If you have a mix of agents with some having [distributed tracing](/docs/apm/distributed-tracing/getting-started/introduction-distributed-tracing) turned on and some having it turned off you will not see the relations between those agents. The solution for this scenario is to run all agents either with distributed tracing turned on (preferred) or turn it off for all agents. If this does not remedy the issue, the service you're trying to view require manual instrumentation.

Author: ```suggestion
<figcaption>
An example of an automap having the instrument option. 
</figcaption>

Make sure that the entities are being monitored with an agent. If you have a mix of agents with some having [distributed tracing](/docs/apm/distributed-tracing/getting-started/introduction-distributed-tracing) turned on and some having it turned off you will not see the relations between those agents. The solution for this scenario is to run all agents either with distributed tracing turned on (preferred) or turn it off for all agents. If this does not remedy the issue, the service you're trying to view requires manual instrumentation.
```

> <figcaption>
> An example of an automap having the instrument option. 
> </figcaption>
> 
> Make sure that the entities are being monitored with an agent. If you have a mix of agents with some having [distributed tracing](/docs/apm/distributed-tracing/getting-started/introduction-distributed-tracing) turned on and some having it turned off you will not see the relations between those agents. The solution for this scenario is to run all agents either with distributed tracing turned on (preferred) or turn it off for all agents. If this does not remedy the issue, the service you're trying to view require manual instrumentation.

Author: ```suggestion
<figcaption>
An example of an automap having the instrument option. 
</figcaption>

Make sure that the entities are being monitored with an agent. If you have a mix of agents with some having [distributed tracing](/docs/apm/distributed-tracing/getting-started/introduction-distributed-tracing) turned on and some having it turned off you will not see the relations between those agents. The solution for this scenario is to run all agents either with distributed tracing turned on (preferred) or turn it off for all agents. If this does not remedy the issue, the service you're trying to view requires manual instrumentation.
```
